tests/package.json

```diff
@@ -20,7 +20,8 @@
     "clean": "rimraf dist fixtures/*/routes.ts fixtures/*/customRoutes.ts",
     "pretest": "yarn clean && cross-env NODE_ENV=tsoa_test ts-node --require tsconfig-paths/register ./prepare.ts",
     "loading-msg-for-tests": "echo '... starting a test run to ensure quality. Note this takes some time as tests do not run in parallel (by design) ...' ",
-    "test": "node ../ensureYarn.js && tsc -b tsconfig.build.json && yarn loading-msg-for-tests && cross-env NODE_ENV=tsoa_test mocha '**/*.spec.ts' --exit --require ts-node/register --require tsconfig-paths/register --timeout 5000",
+    "test": "node ../ensureYarn.js && tsc -b tsconfig.build.json && yarn loading-msg-for-tests && yarn test-esm && cross-env NODE_ENV=tsoa_test mocha '**/*.spec.ts' --exclude 'esm/**/*' --exit --require ts-node/register --require tsconfig-paths/register --timeout 5000",
+    "test-esm": "yarn --cwd esm test",
```

Comment:
Move to the esm pgk.json

Reply:
What do you mean?
it's running the test command from the esm package.json

Reply:
If the workspace is registered properly, lerna should pick it up if you run `test --parallel`.
This is what I wanted to get to, but the top pkg doesn't execute test in all the workspaces, therefore the test command in tests has to call the esm tests.
Instead, change the top level tests to lerna run --parallel test and remove the cross references from here to the esm tests.

Reply:
Oh I understand,
I can't do that with the current implementation because the `pretest` step cleans and builds the required assets for both tests, I don't wanna duplicate the setup and cleanup code.
That means you can't really run it in parallel.

We can copy everything but I'm not sure we should

Reply:
I am under the assumption that lerna will respect dependencies when building the execution plan. I'll try to check myself to verify, but assuming this is indeed the case, if set up correctly, the order should be test.pretest before both test commands. The command then should be `--stream`, not `--parallel though` in order to preserve the topology.

Reply:
Let's move this out, I don't have time to propose a fix and I don't want delay this, the functionality lgtm.